### DISCUSSION TITLE
Input Type is not assignable to type 'undefined'

### DISCUSSION THREAD

**Discussion by srhtak** [trpc/trpc#2473] [Q&A] [answered]
### client

```shell
import { trpc } from "@/utils/trpc";


export default function IndexPage() {
  const hello = trpc.useQuery(['hello',{message:'example'}]); //type '{ message: string; }' is not assignable to type 'undefined'.
  return (
    <div>
      <p>home</p>
    </div>
  );
};

```

### route

```shell 
import { createRouter } from './createRouter';
import {z} from 'zod';



export const appRouter = createRouter().query('hello', {
  input: z.object({
    name: z.string().nullish(),
  }).nullish(),
  resolve({input}){
    return {
      message: `Hello ${input?.name ?? `world`}`,
    };
  }
});

export type AppRouter = typeof appRouter;

```

**Comment by sachinraja** [comment]
I would check to make sure you have strict mode on. If you do, then we'll need a reproduction.

**Reply by srhtak** [reply]
Thank you for your answer. I checked again, strict mode is on but have the same error.

**Reply by KATT** [reply]
We need a repository with a reproduction of this, we can't help with this little info, it's very likely it's an error in your setup.

**Reply by lantos1618** [reply]
I'm also getting this error, think its a TS config. Trying to use with svelte. Will upload an example

**Comment by KATT** [comment, marked as answer]
I think I spotted the issue.

The router has `input`

```ts
z.object({
  name: z.string().nullish(),
}).nullish(),
```

And you're calling it with

```tsx
const hello = trpc.useQuery(['hello', {message:'example'} ]);
```

**It should be:**


```tsx
const hello = trpc.useQuery(['hello', {name:'example'} ]);
```

**Reply by lantos1618** [reply]
Thanks, it was this and also 
"strict" true || "strictNullChecks": true,
in tsconfig.

**Comment by srhtak** [comment]
When I typed name instead of message, I got an error again, then I set up a project from scratch and tried it.  I think it was related to the setup.  Thanks for the answer
